libraries/node-core-library/src/test/LockFile.test.ts

```diff
@@ -164,6 +164,37 @@ describe('LockFile', () => {
 
         const lock: LockFile | undefined = LockFile.tryAcquire(testFolder, resourceName);
 
+        // this lock should be undefined since there is an existing lock
+        expect(lock).toBeUndefined();
+      });
+      test('cannot acquire a lock if another valid lock exists with the same start time', () => {
+        // ensure test folder is clean
+        const testFolder: string = path.join(__dirname, '3');
+        FileSystem.ensureEmptyFolder(testFolder);
+
+        const otherPid: number = 1; // low pid so the other lock is before us
+        const otherPidStartTime: string = '2012-01-02 12:53:12';
+        const thisPidStartTime: string = otherPidStartTime;
+
+        const resourceName: string = 'test';
+
+        const otherPidLockFileName: string = LockFile.getLockFilePath(testFolder, resourceName, otherPid);
+
+        setLockFileGetProcessStartTime((pid: number) => {
+          return pid === process.pid ? thisPidStartTime : otherPidStartTime;
+        });
+
+        // create an open lockfile
+        const lockFileHandle: FileWriter = FileWriter.open(otherPidLockFileName);
+        lockFileHandle.write(otherPidStartTime);
```

Comment:
Will this test work correctly on Windows/Mac/Linux OS's?

Reply:
The test is within the mac/linux ifdef block above already so should be fine.

Reply:
Cool. I'll test your branch on my Windows machine just to be sure, then we can merge it. 👍

Reply:
✔ builds successfully on Windows